pkg/virt-controller/watch/migration.go

```diff
@@ -723,16 +737,33 @@ func (c *MigrationController) handleTargetPodHandoff(migration *virtv1.VirtualMa
 		c.recorder.Eventf(migration, k8sv1.EventTypeWarning, FailedHandOverPodReason, fmt.Sprintf("Failed to set MigrationStat in VMI status. :%v", err))
 		return err
 	}
+
 	log.Log.Object(vmi).Infof("Handed off migration %s/%s to target virt-handler.", migration.Namespace, migration.Name)
 	c.recorder.Eventf(migration, k8sv1.EventTypeNormal, SuccessfulHandOverPodReason, "Migration target pod is ready for preparation by virt-handler.")
+
 	return nil
 }
 
 func (c *MigrationController) handleSignalMigrationAbort(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance) error {
 
 	vmiCopy := vmi.DeepCopy()
+	now := v1.NewTime(time.Now())
+	isMigrationAborted := c.isMigrationAborted(migration.UID)
 
+	if vmiCopy.Status.MigrationState == nil {
+		vmiCopy.Status.MigrationState = &virtv1.VirtualMachineInstanceMigrationState{
+			MigrationUID:   migration.UID,
+			SourceNode:     vmi.Status.NodeName,
+			StartTimestamp: &now,
+		}
+	}
+
+	if vmiCopy.Status.MigrationState.EndTimestamp == nil && isMigrationAborted {
+		vmiCopy.Status.MigrationState.EndTimestamp = &now
+		vmiCopy.Status.MigrationState.Failed = true
```

Comment:
virt-controller does not own this field. This is virt-handler's responsibility. The only time virt-controller should set the EndTimestamp or Failed fields on the migration object is if the target pod disappeared and we're 100% certain that there is no migration taking place within virt-handler.

Reply:
> The only time virt-controller should set the EndTimestamp or Failed fields on the migration object is if the target pod disappeared and we're 100% certain that there is no migration taking place within virt-handler.

This is in fact exactly the case here. Let me try to clarify.

This change (and the whole PR in general) aims to tackle the scenario in which the migration is cancelled before the hand-off occurs. In such a case it's guaranteed that the source virt-handler did not start the migration process. To identify such migrations, I've introduced the `abortedMigrations` map. This map holds migration UIDs for migrations that are aborted but weren't handed-off yet.

The idea is to take advantage of the fact the no hand-off occurred and avoid it. Instead, we'd like the controller to fail the migration without involving virt-handler.

This is important both for performance reasons (we shouldn't hand-off and then wait for the virt-handler to abort the migration. Instead we should abort it before handing off) and for situations in which the target pod is unschedulable (in this case the migration will get stuck until the target pod would be timed out, like happened [here](https://bugzilla.redhat.com/show_bug.cgi?id=1719190)).

Reply:
Small Note:
Even though the migration is not handed-off yet, the target pod might be already created. If this is the case, we should try to delete it. We may miss the pod's creation (i.e. the pod can be created right after we fail the migration), but this is okay since in such cases the target would eventually die because of time-out. This corner case is much better than having to wait for a target pod to time out if the target pod is unschedulable, but we should still avoid it if we can. That's why I've already put out [this PR](https://github.com/kubevirt/kubevirt/pull/7787) to better handle these corner cases by trying to clean up "orphan migration pods".